com.avaloq.tools.ddk.xtext.ide/src/com/avaloq/tools/ddk/xtext/ide/findrefs/InferredModelReferenceFilter.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Group AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Group AG - initial API and implementation
+ *******************************************************************************/
+package com.avaloq.tools.ddk.xtext.ide.findrefs;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.xtext.resource.IReferenceDescription;
+
+import com.google.common.base.Predicate;
+
+
+/**
+ * A filter for filtering out inferred model elements.
+ */
+public class InferredModelReferenceFilter implements Predicate<IReferenceDescription> {
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return true if {@code input} is not an inferred model element.
+   */
+  @Override
+  public boolean apply(final IReferenceDescription input) {
+    return !isInferredModelElement(input.getSourceEObjectUri());
+  }
+
+  /**
+   * Checks a model element's URI to determine if it is an inferred model element.
+   *
+   * @param elementURI
+   *          the URI to check
+   * @return true if {@code elementURI} is the URI of an inferred model element; false otherwise.
+   */
+  protected boolean isInferredModelElement(final URI elementURI) {
+    String fragment = elementURI.fragment();
+    if (fragment.length() > 1) {
+      char firstCharacter = fragment.charAt(1);
+      // Normally, URI fragments of non-inferred elements look like: /0/x/y/z/

```

Comment:
We no longer have a custom FragmentProvider for WFD, it was used for the graphical editor and we have removed both. Have I missed something?

Reply:
This is just a copy of com.avaloq.tools.ddk.xtext.ui.editor.findrefs.InferredModelReferenceFilter - I have made no other changes. That said, even if the specific case in the comment is no longer present, shouldn't we still cope with anyone else that writes a custom uri fragment provider? (I think we should still change the comment though)

Reply:
I could only find two in our codebase (IntfDef, CodeTab), and I am not sure, but I think they all look like /0/x/y/z, so we would not need the exception. Could you check?